client.go

```diff
@@ -2907,3 +2902,75 @@ func releasePipelineWork(pool *sync.Pool, w *pipelineWork) {
 	w.err = nil
 	pool.Put(w)
 }
+
+// Resource Clean interface.
+// resourceClean can register into clientCleaner for cleaning resources.
+type resourceClean interface {
```

Comment:
Maybe call it `cleanableClient`? Either you use `resource` everywhere and make it all generic. Or use `client` and make it specific for clients.